frontend/src/hooks.tsx

```diff
@@ -0,0 +1,172 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
+import type { RootState, AppDispatch } from './store'
+import { FMSId, FMSSampleNextStepByStudy, FMSStudy, WorkflowStepOrder } from './models/fms_api_models'
+import { ReactElement, useCallback, useEffect, useState } from 'react'
+import { SampleAndLibrary } from './components/WorkflowSamplesTable/ColumnSets'
+import { fetchSamplesAndLibraries } from './modules/studySamples/services'
+import { Protocol, Sample } from './models/frontend_models'
+import api from './utils/api'
+import React from 'react'
+import { Popover, Spin, Tag } from 'antd'
+import { fetchStudies } from './modules/cache/cache'
+
+// Use throughout your app instead of plain `useDispatch` and `useSelector`
+export const useAppDispatch: () => AppDispatch = useDispatch
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
+
+export function useSampleAndLibraryList(sampleIDs: FMSId[], offset?: number, pageSize?: number) {
```

Comment:
you want to do away with the component structure for hooks ? These do not look like they are widely applicable in the app.

Reply:
Are you sure you want your hooks to return JSX elements? This seem to me that this may limit the reusability of your hooks.

Reply:
I've decided to undo that unnecessary change and a few others not related to labwork samples